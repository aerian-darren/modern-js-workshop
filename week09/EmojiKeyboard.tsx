import * as React from "react";
import { EmojiKey } from "./EmojiKey";
import { Emoji } from "./Emoji";

interface Props {
    emojis: Emoji[];
    onAddEmoji: (moji: string) => void;
}
export const EmojiKeyboard: React.SFC<Props> = ({ emojis, onAddEmoji }) => (
    <div id="keyboard">
        {emojis.map((emoji) => (
            <EmojiKey key={emoji.code} emoji={emoji} onAddEmoji={onAddEmoji} />
        ))}
    </div>
);
