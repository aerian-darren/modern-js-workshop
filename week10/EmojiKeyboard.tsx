import * as React from "react";
import { Emoji } from "./Emoji";
import { EmojiKey } from "./EmojiKey";

interface Props {
    emojis: Emoji[];
    onAddEmoji: (moji: string) => void;
}
export const EmojiKeyboard: React.SFC<Props> = ({ emojis, onAddEmoji }) => (
    <div id="keyboard">
        {emojis.map((emoji) => (
            <EmojiKey key={emoji.code} emoji={emoji} onAddEmoji={onAddEmoji} />
        ))}
    </div>
);
