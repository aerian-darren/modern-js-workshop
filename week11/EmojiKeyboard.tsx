import * as React from "react";
import { Emoji } from "./Emoji";
import { EmojiKey } from "./EmojiKey";

interface Props {
    emojis: Emoji[];
    onAddEmoji: (moji: string) => void;
    clickedKeys: string[];
}
export const EmojiKeyboard: React.SFC<Props> = ({
    emojis,
    onAddEmoji,
    clickedKeys,
}) => (
    <div id="keyboard">
        {emojis.map((emoji: Emoji) => (
            <EmojiKey
                key={emoji.code}
                emoji={emoji}
                onAddEmoji={onAddEmoji}
                disabled={
                    clickedKeys.filter((moji) => moji === emoji.moji).length >=
                    3
                }
            />
        ))}
    </div>
);
