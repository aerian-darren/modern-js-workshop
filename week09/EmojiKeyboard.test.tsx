import * as React from "react";
import { EmojiKeyboard } from "./EmojiKeyboard";
import * as renderer from "react-test-renderer";
import emojilist from "../lib/emoji";

it("renders correctly", () => {
    const tree = renderer
        .create(<EmojiKeyboard emojis={emojilist} onAddEmoji={() => {}} />)
        .toJSON();

    expect(tree).toMatchSnapshot();
});
