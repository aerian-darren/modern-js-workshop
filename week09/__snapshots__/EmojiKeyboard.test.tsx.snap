// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<div
  id="keyboard"
>
  <button
    onClick={[Function]}
  >
    🎿
  </button>
  <button
    onClick={[Function]}
  >
    😻
  </button>
  <button
    onClick={[Function]}
  >
    😿
  </button>
  <button
    onClick={[Function]}
  >
    🙋
  </button>
  <button
    onClick={[Function]}
  >
    🙌
  </button>
  <button
    onClick={[Function]}
  >
    🚁
  </button>
  <button
    onClick={[Function]}
  >
    🏃
  </button>
  <button
    onClick={[Function]}
  >
    🏷
  </button>
  <button
    onClick={[Function]}
  >
    👃
  </button>
  <button
    onClick={[Function]}
  >
    🗢
  </button>
  <button
    onClick={[Function]}
  >
    👑
  </button>
  <button
    onClick={[Function]}
  >
    👕
  </button>
  <button
    onClick={[Function]}
  >
    👖
  </button>
  <button
    onClick={[Function]}
  >
    👗
  </button>
  <button
    onClick={[Function]}
  >
    👛
  </button>
  <button
    onClick={[Function]}
  >
    👝
  </button>
  <button
    onClick={[Function]}
  >
    👢
  </button>
  <button
    onClick={[Function]}
  >
    👦
  </button>
  <button
    onClick={[Function]}
  >
    👧
  </button>
  <button
    onClick={[Function]}
  >
    👶
  </button>
  <button
    onClick={[Function]}
  >
    👻
  </button>
  <button
    onClick={[Function]}
  >
    👼
  </button>
  <button
    onClick={[Function]}
  >
    👽
  </button>
  <button
    onClick={[Function]}
  >
    💀
  </button>
  <button
    onClick={[Function]}
  >
    💊
  </button>
  <button
    onClick={[Function]}
  >
    💍
  </button>
  <button
    onClick={[Function]}
  >
    💎
  </button>
  <button
    onClick={[Function]}
  >
    💴
  </button>
  <button
    onClick={[Function]}
  >
    💶
  </button>
  <button
    onClick={[Function]}
  >
    💷
  </button>
  <button
    onClick={[Function]}
  >
    💺
  </button>
  <button
    onClick={[Function]}
  >
    📀
  </button>
  <button
    onClick={[Function]}
  >
    📅
  </button>
  <button
    onClick={[Function]}
  >
    📔
  </button>
  <button
    onClick={[Function]}
  >
    📖
  </button>
  <button
    onClick={[Function]}
  >
    📚
  </button>
  <button
    onClick={[Function]}
  >
    📟
  </button>
  <button
    onClick={[Function]}
  >
    📠
  </button>
  <button
    onClick={[Function]}
  >
    📣
  </button>
  <button
    onClick={[Function]}
  >
    📻
  </button>
  <button
    onClick={[Function]}
  >
    📼
  </button>
  <button
    onClick={[Function]}
  >
    🔑
  </button>
  <button
    onClick={[Function]}
  >
    🔒
  </button>
  <button
    onClick={[Function]}
  >
    🔔
  </button>
  <button
    onClick={[Function]}
  >
    🕳
  </button>
  <button
    onClick={[Function]}
  >
    🖪
  </button>
  <button
    onClick={[Function]}
  >
    🖫
  </button>
  <button
    onClick={[Function]}
  >
    🖺
  </button>
  <button
    onClick={[Function]}
  >
    🗈
  </button>
  <button
    onClick={[Function]}
  >
    🗏
  </button>
  <button
    onClick={[Function]}
  >
    🗐
  </button>
  <button
    onClick={[Function]}
  >
    🗿
  </button>
  <button
    onClick={[Function]}
  >
    😁
  </button>
  <button
    onClick={[Function]}
  >
    😃
  </button>
  <button
    onClick={[Function]}
  >
    😄
  </button>
  <button
    onClick={[Function]}
  >
    😉
  </button>
  <button
    onClick={[Function]}
  >
    😏
  </button>
  <button
    onClick={[Function]}
  >
    😓
  </button>
  <button
    onClick={[Function]}
  >
    😠
  </button>
  <button
    onClick={[Function]}
  >
    😡
  </button>
  <button
    onClick={[Function]}
  >
    😩
  </button>
  <button
    onClick={[Function]}
  >
    😪
  </button>
  <button
    onClick={[Function]}
  >
    😯
  </button>
  <button
    onClick={[Function]}
  >
    😱
  </button>
  <button
    onClick={[Function]}
  >
    😷
  </button>
  <button
    onClick={[Function]}
  >
    🚪
  </button>
  <button
    onClick={[Function]}
  >
    🛀
  </button>
  <button
    onClick={[Function]}
  >
    ⌚️
  </button>
</div>
`;
